src/common/datatypes/Geography.cpp

```diff
@@ -0,0 +1,164 @@
+/* Copyright (c) 2020 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "common/datatypes/Geography.h"
+
+#include <folly/String.h>
+#include <folly/hash/Hash.h>
+#include <s2/s2loop.h>
+#include <s2/s2polygon.h>
+
+#include <cstdint>
+
+#include "common/geo/io/wkb/WKBReader.h"
+#include "common/geo/io/wkt/WKTWriter.h"
+
+namespace nebula {
+
+GeoShape Geography::shape() const {
+  // TODO(jie) May store the shapetype as the data member of Geography is ok.
+  std::string wkbCopy = wkb;
+  uint8_t* beg = reinterpret_cast<uint8_t*>(wkbCopy.data());
+  uint8_t* end = beg + wkbCopy.size();
+  WKBReader reader;
+  auto byteOrderRet = reader.readByteOrder(beg, end);
+  DCHECK(byteOrderRet.ok());
+  ByteOrder byteOrder = byteOrderRet.value();
+  auto shapeTypeRet = reader.readShapeType(beg, end, byteOrder);
+  DCHECK(shapeTypeRet.ok());
+  return shapeTypeRet.value();
+}
+
+std::string Geography::asWKT() const {
+  auto geomRet = WKBReader().read(wkb);
+  DCHECK(geomRet.ok());
+  std::unique_ptr<Geometry> geom = std::move(geomRet).value();
+  std::string wkt = WKTWriter().write(geom.get());
+  LOG(INFO) << "Geography::asWKT(): " << wkt;
+  return wkt;
+}
+
+std::string Geography::asWKB() const { return folly::hexlify(wkb); }
+
+std::unique_ptr<S2Region> Geography::asS2() const {
+  auto geomRet = WKBReader().read(wkb);
+  DCHECK(geomRet.ok());
```

Comment:
ditto. Remember you should ensure that nebula graph could not crash since the user's  error input.